### ISSUE TITLE
turn process CPU 100%

### ISSUE THREAD

**Issue by clia** [webrtc-rs/turn#4]
Soundly the turn server received some bytes not conforming the standards, then the process always running CPU in 100%.
The server log is:

```
listening 0.0.0.0:20011...
Waiting for Ctrl-C...
[2021-04-08T23:13:45Z DEBUG webrtc_turn::server::request] received 407 bytes of udp from 193.107.216.134:5088 on 0.0.0.0:20011
[2021-04-08T23:13:45Z DEBUG webrtc_turn::server::request] handle_turn_packet
[2021-04-08T23:13:45Z ERROR webrtc_turn::server] error when handling datagram: 4f4e5320 is invalid magic cookie (should be 2112a442)
```

Version: webrtc-turn v0.1.2

**Comment by rainliu** [comment]
Thanks for reporting this issue. Will take a look soon.

**Comment by clia** [comment]
Looks likely the problem exists always, when started the example server, it appears.

```
   Compiling webrtc-turn v0.1.3 (/Users/clia/github.com/webrtc-rs/turn)
    Finished dev [unoptimized + debuginfo] target(s) in 43.97s
     Running `target/debug/examples/turn_server_udp --public-ip 0.0.0.0 --users user=pass`
[2021-04-09T03:04:21Z TRACE mio::poll] registering event source with poller: token=Token(1), interests=READABLE | WRITABLE
listening 0.0.0.0:3478...
Waiting for Ctrl-C...
```

**Comment by rainliu** [comment]
@clia, looks like it is due to the following code piece. I need to figure out a better way to handle Ctrl-C signal.

```
let running = Arc::new(AtomicBool::new(true));
    let r = running.clone();

    ctrlc::set_handler(move || {
        r.store(false, Ordering::SeqCst);
    })
    .expect("Error setting Ctrl-C handler");

    println!("Waiting for Ctrl-C...");
    while running.load(Ordering::SeqCst) {}
    println!("\nClosing connection now...");
```

**Comment by rainliu** [comment]
fixed this issue in commit: e14fbebac58f0f03fe7a08cc2c6f96425317c285

**Comment by clia** [comment]
👍
